### PR TITLE
Block: use `wp.date.getSettings` if available

### LINKED ISSUE #12787: Block: use `wp.date.getSettings` if available
## Task Description

> `wp.date.__experimentalGetSettings` is deprecated since version 6.1. Please use `wp.date.getSettings` instead

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Create a post. 
2. Add a block. 
3. Select Selected Stories
4. Select Circle Carousel
5. Select "Select Stories"
6. See Select Stories working correctly. Check console see no errors or notices. 

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12787

### CODE REVIEW

File: packages/stories-block/src/block/components/storyCard.js [hunk 1]
```diff
@@ -45,7 +50,8 @@ function StoryCard({
   const hasContentOverlay = isShowingTitle || isShowingAuthor || isShowingDate;
 
   // @todo Keep an eye on this experimental API, make necessary changes when this gets updated in core.
```
Comment (swissspidy) [inline packages/stories-block/src/block/components/storyCard.js]: This todo is now outdated

File: packages/stories-block/src/block/components/storyPicker/storyPreview.js [hunk 2]
```diff
@@ -41,7 +45,8 @@ function StoryPreview({
   isSelectable = true,
 }) {
   // @todo Keep an eye on this experimental API, make necessary changes when this gets updated in core.
```
Comment (swissspidy) [inline packages/stories-block/src/block/components/storyPicker/storyPreview.js]: This todo is now outdated

File: packages/stories-block/src/block/components/storyCard.js (hunk 1 above)
Comment (spacedmonkey) [reply packages/stories-block/src/block/components/storyCard.js]: Updated todo.

File: packages/stories-block/src/block/components/storyPicker/storyPreview.js (hunk 2 above)
Comment (spacedmonkey) [reply packages/stories-block/src/block/components/storyPicker/storyPreview.js]: Updated.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**